pyiron/dft/job/generic.py

```diff
@@ -338,6 +386,37 @@ def get_electronic_structure(self):
                 es_obj.from_hdf(ho)
             return es_obj
 
+    def modify_kpoints(self):
+        if self.k_mesh_per_reciprocal_angstrom is not None:
+            self.set_kpoints(center_shift=self.k_mesh_center_shift,
+                             kpoints_per_reciprocal_angstrom=self.k_mesh_per_reciprocal_angstrom,
+                             symmetry_reduction=self.reduce_kpoint_symmetry)
+
+
+def get_k_mesh_by_density(cell, kmesh_density_per_inverse_angstrom=1.0):
```

Comment:
Is there a particular reason that `kmesh_density_per_inverse_angstrom` has a default value?

Reply:
No particular reason since this is a helper function @yury-lysogorskiy wrote. The function within the class does not have a default value.

Reply:
Then isn't it better to remove the default value and set the same order as in `get_k_mesh_by_cell`?

Reply:
Now, from my previous practical experience, I would recommend default value of `kmesh_density_per_inverse_angstrom=0.5`, even if that would be overwritten by the external calling method...

Reply:
And if we still could make a name-changes, I found a better name for that parameter: `kmesh_spacing`

Reply:
I'm fine with giving a default value to k point mesh or k point density, but I think we shouldn't do both at the same time. And if we don't give a default value in one place, it shouldn't appear in other places either for the same argument. This being said, I would have preferred @yury-lysogorskiy's suggestion to set a default value for the k point density (over setting k-point mesh), but in the current implementation in pyiron it's probably not a very practical solution.

And I agree with @yury-lysogorskiy on the name change. Rather than making the description of what it does accurately, it's important that the user understands clearly whether an increasing `k_xxx` leads to an increasing number of k points or other way around. In this regard, I also think `spacing` does a good job.

Reply:
I've made the name change. By default `k_mesh_spacing` is `None` which means whatever k-point setting is the default (4x4x4 I think) will be used. Only when you set `k_mesh_spacing`  explicitly or set it through `set_kpoints()` does this value change based on cell shape.